internal/binutils/binutils.go

```diff
@@ -140,7 +144,87 @@ func initTools(b *binrep, config string) {
 		b.addr2line, b.addr2lineFound = findExe("gaddr2line", append(paths["addr2line"], defaultPath...))
 	}
 	b.nm, b.nmFound = findExe("nm", append(paths["nm"], defaultPath...))
-	b.objdump, b.objdumpFound = findExe("objdump", append(paths["objdump"], defaultPath...))
+	// MacOS binary is named objdump, Linux binary is named llvm-objdump-version
+	b.objdump, b.llvmObjdumpFound, b.objdumpFound = findObjdump(append(paths["llvm-objdump"], append(paths["objdump"], defaultPath...)...))
+}
+
+// findObjdump finds all binaries named objdump, llvm-objdump*. If multiple LLVM objdump are found,
+// it picks objdump with the latest major version. If no LLVM objdump is found, returns the path to
+// any binary named objdump, if found.
+func findObjdump(paths []string) (string, bool, bool) {
+	allObjdump := findAllObjdump("llvm-objdump*", paths)
+
+	if objdump, objdumpFound := findExe("llvm-objdump", paths); objdumpFound {
+		allObjdump = append(allObjdump, objdump)
+	}
+
+	if objdump, objdumpFound := findExe("llvm-objdump-9", paths); objdumpFound {
+		allObjdump = append(allObjdump, objdump)
+	}
+
+	buObjdumpFound := false
+	var buObjdump string
+	if objdump, objdumpFound := findExe("objdump", paths); objdumpFound {
+		cmdOut, _ := exec.Command(objdump, "--version").Output()
+		if strings.Contains(string(cmdOut), "GNU") && strings.Contains(string(cmdOut), "Binutils") {
```

Comment:
It may be slightly more robust to check just the first line (`GNU objdump (GNU Binutils for Debian) 2.34`)

Reply:
The regex is also not searching only on the first line, so functionally these two seem similar. We also do not want to search only first line. For example, `<objdump> --version` prints version in different lines on mac & linux